paasta_tools/cli/cmds/spark_run.py

```diff
@@ -838,9 +866,9 @@ def configure_and_run_docker_container(
     if pod_template_path:
         volumes.append(f"{pod_template_path}:{pod_template_path}:rw")
 
-    volumes.append(
-        f"{system_paasta_config.get_spark_kubeconfig()}:{system_paasta_config.get_spark_kubeconfig()}:ro"
-    )
+    # Mounting directory because there are multiple KUBECONFIGs due to migration
```

Comment:
i'd argue that the more important bit is that there's a helper script we need included - otherwise we'd conditionally mount one file or the other :p
```suggestion
    # NOTE: we mount a directory here since the kubeconfig we're transitioning to requires a helper script that will co-exist in the same directory
```

Reply:
(that said, as i mentioned in the other comment: it does seem somewhat brittle to depend on both kubeconfigs existing in the same directory - but probably fine if we're only going to have 2 kubeconfigs for a short period of time)